### PR TITLE
Feat/keycloak logout support

### PR DESCRIPTION
**Pull request by renaud-twd**
Hi,
First of all, thanks for this awesome package that greatly helps dealing with OpenID Connect and saved me a big amount of work !!!

I have an issue with the package that fails loging the user out with Keycloak ID provider.
Thus, I added propose you this PR to fix this :
* Add capabilities to logout a keycloak user
* Also add the capabilities to revoke a keycloak token

> All these changes are backward compatible.

(By the way, I wonder if there is a bug in your logout code : you use `get` request instead of a `post` request + you should probably clear user's identity)

Thanks again for sharing this code.
Best regards,
Renaud

### CODE REVIEW

**Comment by jhancock4d** [comment]
Hi, sorry I've been away from this project so I didn't even see these (and github didn't notify me for some reason).

Could you update the code?  Generally the default is GET for logout but can be POST per the spec too, so the solution needs to read this from the site's documentation for what method to use.

Thanks!

**Comment by renaud-twd** [comment]
Hi @jhancock4d,

No worries, thanks for this (late) feedback. ;)
I just updated the code with solved conflicts (which were located in `pubspec.lock` files)

File: openidconnect/lib/src/models/requests/revoke_token_request.dart [hunk 1]
```diff
@@ -6,17 +6,25 @@ class RevokeTokenRequest {
...
-            ? "access_token"
-            : "refresh_token",
-      };
+  Map<String, String> toMap() {
+    var map = {"token": token,
+      "token_type_hint": tokenType == TokenType.accessToken
+          ? "access_token"
+          : "refresh_token",
+    };
+    if (clientId != null) map = {"client_id": clientId!, ...map};
```
Comment (ahmednfwela) [inline openidconnect/lib/src/models/requests/revoke_token_request.dart:26]: why not just add these in the original map ?
```dart
 var map = {"token": token,
      "token_type_hint": tokenType == TokenType.accessToken
          ? "access_token"
          : "refresh_token",
       if (clientId != null) "client_id": clientId,
       if (clientSecret != null) "client_secret": clientSecret
    };
```

File: openidconnect/lib/src/openidconnect_client.dart [hunk 2]
```diff
@@ -368,18 +405,21 @@ class OpenIdConnectClient {
...
-      }
+      clearIdentity();
+      _raiseEvent(AuthEvent(AuthEventTypes.Error, message: e.toString()));
       return false;
     } finally {
       _refreshing = false;
     }
   }
 
+  void clearIdentity() async {
```
Comment (ahmednfwela) [inline openidconnect/lib/src/openidconnect_client.dart:416]: i think this should be a Future<void>

Comment (renaud-twd) [reply openidconnect/lib/src/openidconnect_client.dart]: @ahmednfwela thanks for the feedback, it is fixed !

File: openidconnect/lib/src/models/requests/revoke_token_request.dart (hunk 1 above)
Comment (renaud-twd) [reply openidconnect/lib/src/models/requests/revoke_token_request.dart:26]: @ahmednfwela thanks for the feedback, I just did it the way it was done everywhere else in this codebase. :)

Let me know if I need to change it.

Comment (jhancock4d) [reply openidconnect/lib/src/models/requests/revoke_token_request.dart:26]: FYI, the reason I hid the original map is specifically to protect against inadvertent overwriting of the properties that have to be there.

Comment (ahmednfwela) [reply openidconnect/lib/src/models/requests/revoke_token_request.dart:26]: I did a major refactor for readability, and changed the entire package to use this pattern
```dart
var map = {
      "token": token,
      "token_type_hint": tokenType == TokenType.accessToken
          ? "access_token"
          : "refresh_token",
       if (clientId != null) "client_id": clientId,
       if (clientSecret != null) "client_secret": clientSecret
    };
```
so please change it in the code

**Comment by ahmednfwela** [comment]
related to https://github.com/4D-Technologies/openidconnect_flutter/issues/39 as well

File: openidconnect/lib/src/models/requests/revoke_token_request.dart (hunk 1 above)
Comment (renaud-twd) [reply openidconnect/lib/src/models/requests/revoke_token_request.dart:26]: @ahmednfwela hello, sorry but it looks like this PR has been merged and closed.

Comment (ahmednfwela) [reply openidconnect/lib/src/models/requests/revoke_token_request.dart:26]: no problem 👍 will do the refactor myself

**Pull request merged**